mlir/lib/Analysis/DataFlow/IntegerRangeAnalysis.cpp

```diff
@@ -45,9 +45,13 @@ void IntegerValueRangeLattice::onUpdate(DataFlowSolver *solver) const {
   std::optional<APInt> constant = getValue().getValue().getConstantValue();
   auto value = cast<Value>(anchor);
   auto *cv = solver->getOrCreateState<Lattice<ConstantValue>>(value);
-  if (!constant)
-    return solver->propagateIfChanged(
-        cv, cv->join(ConstantValue::getUnknownConstant()));
+  if (!constant) {
```

Comment:
This is actually the intended usage of the `propagateIfChanged` method. Data flow variables that depend on each other can issue updates to each other through this method.